css: add object-view-box property page (#39934)
mdn/content: files/en-us/web/css/object-view-box/index.md

  }
}
```

## Syntax

```css
object-view-box: none;

/* inset(allSides) */
object-view-box: inset(20%);
/* inset(vertical horizontal) */
object-view-box: inset(20% 30%);
/* inset(top right bottom left) */
object-view-box: inset(10px 0 25px 33px);

/* xywh(x y width height) */
object-view-box: xywh(95px 20px 60px 60px);

/* rect(top right bottom left) */
object-view-box: rect(10px 30px 30px 10px);

/* Global values */
object-fit: inherit;
object-fit: initial;
object-fit: revert;
object-fit: revert-layer;
object-fit: unset;
```

### Values

- `none`

  - : No view box is specified. It makes the entire content visible.

- `<basic-shape-rect>`
  - : Specifies a view box for the element using {{cssxref("basic-shape/inset","inset()")}}, {{cssxref("basic-shape/xywh","xywh()")}}, or {{cssxref("basic-shape/rect","rect()")}} function.

## Formal definition

{{cssinfo}}

## Formal syntax

{{csssyntax}}

## Examples

### Live zoom in using object-view-box property

In this example, we allow users to zoom in on a leopard image. The zoom-in point is fixed at the eye of the leopard. We take user input using HTML [`range`](Web/HTML/Reference/Elements/input/range) input element.

> ```html

Author: ```suggestion
### HTML

We include an {{htmlelement("img")}} and a [`range`](Web/HTML/Reference/Elements/input/range) {{htmlelement("input")}} element, associating a {{htmlelement("label")}} with the `<input>` .

```html
```

<img
  src="https://mdn.github.io/shared-assets/images/examples/leopard.jpg"
  alt="leopard" />
<br />
<label for="box-size">Zoom: </label>

> <input type="range" id="box-size" min="100" max="350" value="150" />

Author: Steps appear weird when the bar is moved using the mouse. Sudden changes in image are not good for eyes.

```

> 

Author: ```suggestion

### CSS
```

```css hidden
input {
  width: 350px;
}

@supports not (object-view-box: none) {
  body::before {
    content: "The 'object-view-box' property is not supported by your browser.";
    color: black;
    background-color: #ffcd33;
    display: block;
    width: 100%;
    text-align: center;
  }

>   body > * {

Author: good point

    display: none;
  }
}
```

```css
:root {
  --box-size: 350px;
}

img {
  width: 450px;
  height: 350px;
  object-view-box: xywh(500px 30px var(--box-size) var(--box-size));
}
```

In the above CSS, we define a custom property `--box-size` to adjust the view box size. Decreasing the size of the view box gives a zoom-in effect. Increasing the view box size gives a zoom-out effect. The `xywh()` function accepts an offset point and size of the view box. The view box's offset point is set at (500px, 30px), which is the top left corner of the right eye of the leopard.

> 

Author: ```suggestion

### JavaScript

```

![An image demonstrating extrinsic and intrinsic image sizes](extrinsic-intrinsic_sizes.jpg)

The `object-view-box` property is more flexible than the `object-fit` property, and it is capable of doing more things. For example, it can be used to crop, zoom, and pan images. The property sets viewable area (view box), which defines what part of the content to show and how to fit it inside the extrinsic size. The view box value contains a rectangle and its position relative to the intrinsic area of the content, but _physical size of the view box remains equal to the extrinsic size_. The view box marks the area in the content to be displayed, and then it is transformed to match the extrinsic dimensions fitting into the HTML element. In the following image, we have the same leopard picture in a 400 x 150 image element. However, this time we have used `object-view-box` property to crop the leopard's eye portion of the picture.

> ![An image showing leopard eyes cropped from an leopard image using object-view-box property](https://mdn.github.io/shared-assets/images/diagrams/css/object-view-box/object-view-box_xywh.jpg)

Reviewer: ```suggestion
![The leopard image cropped using object-view-box property, with a 400px by 150px viewbox displaying an unscaled section of the image](https://mdn.github.io/shared-assets/images/diagrams/css/object-view-box/object-view-box_xywh.jpg)
```

> Notice, in this case, the extrinsic size and view box size have been kept the same, i.e., 400 x 150 pixels. In other words, the aspect ratios have been kept the same. Maintaining the same aspect ratio prevents image distortion.
> 
> Various image operations can be achieved by having different extrinsic and view box sizes.

Reviewer: ```suggestion
As the rectangle defined by the `object-view-box` property and the rectangle viewbox of the `<img>` element are the same size, i.e., 400 x 150 pixels, the aspect ratios of both are the same and the replaced element is not scaled. 

Maintaining the same aspect ratio prevents image distortion. With `object-view-box`, we can accomplish various image operations while having different extrinsic and view box sizes, without distorting the replaced element as it scales up and down.
```
